### DISCUSSION TITLE
Render merged values from a release template, then run a values template against that before the final values are rendered

### DISCUSSION THREAD

**Discussion by daveneeley** [helmfile/helmfile#1261] [Questions]
https://github.com/roboll/helmfile/issues/1143 seems to describe a similar issue, but it was written prior to the `inherits` syntax and I'm not sure it would work.

I have a helmfile with two bases and two charts.
- *base 1* - loads common values for both charts.
- *base 2* - in a release template
  - overrides common values with environment specific ones
  - loads release-specific files if they exist
  - sets `installedTemplate`, which can turn a release on or off

Both charts inherit the release template. Both have the same value structure. which isn't namespaced. Loading both values files in a shared base could have undesirable merge side effects.

A release template seems like the right thing to use in *base 2*. The downside is that *base 2* only renders a template. The values themselves aren't merged yet. When the final release is executed, `templates[].installedTemplate` and `templates[].valuesTemplate` will be rendered using `.Values` from *base 1*.

Is it possible to render another template after the inherited release template values are merged?

**Comment by yxxhero** [comment]
@daveneeley could you post a demo?

**Reply by daveneeley** [reply]
Sure! Here's a gist https://gist.github.com/daveneeley/5f88d46d4c1f0491a6abdc9af0162a1e

It's contrived but reduces the problem to the simplest form.

**Reply by daveneeley** [reply]
I'm sure it would have a performance impact but I would love to see `.Values` be merged "as you go" and always be available to the next file that is loaded, even in the same phase.
